### PR TITLE
frh repo:294 cant add a repo with same url anymore for local or global, also import checks for an existing repo url and use this url instead to add a ne repo

### PR DESCRIPTION
**Pull request by daniwe4**
https://github.com/conceptsandtraining/doil/issues/294

### CODE REVIEW

File: app/src/Commands/Pack/ImportCommand.php [hunk 1]
```diff
@@ -123,17 +123,21 @@ public function execute(InputInterface $input, OutputInterface $output) : int
                 ->withUrl($project_config->getRepositoryUrl())
             ;
 
-            if ($this->repo_manager->repoExists($repo)) {
-                $this->writer->error(
-                    $output,
-                    "Repository {$repo->getName()} already exists!",
-                    "Use <fg=gray>doil repo:list</> to see current installed repos"
-                );
+            if ($this->repo_manager->repoUrlExists($repo)) {
```
Comment (tbongers-cat) [inline app/src/Commands/Pack/ImportCommand.php:126]: As the new empty repo is generated as local by default, only **local** repositories are checked for the repository url. If the imported instance has a repository url that already exists globally, a new `<instance>_import` repository will still be created locally.

Wouldn't it be better to also check for existing **global** repositories and re-use these as well (e.g. the ILIAS repo)?

**Review by tbongers-cat — COMMENTED** [review]
See comment please

File: app/src/Commands/Pack/ImportCommand.php (hunk 1 above)
Comment (daniwe4) [reply app/src/Commands/Pack/ImportCommand.php:129]: done!

File: app/src/Commands/Repo/RepoManager.php [hunk 2]
```diff
@@ -156,13 +173,30 @@ public function getLocalRepo(string $name) : Repo
         });
 
         if (! count($repos)) {
-            throw new RuntimeException("Repo $name not found in local repos.");
+            throw new RuntimeException("Repo with name '$name' not found in local repos.");
```
Comment (tbongers-cat) [inline app/src/Commands/Repo/RepoManager.php:176]: This is slightly different to the message in getGlobalRepoByName(), might be better if they are both phrased the same way (also needs to be changed in the test result).

File: app/src/Commands/Pack/ImportCommand.php [hunk 3]
```diff
@@ -123,17 +123,36 @@ public function execute(InputInterface $input, OutputInterface $output) : int
...
-                );
+            $local_repo = null;
+            $global_repo = null;
+            $from_global_repo = false;
+            if ($this->repo_manager->repoUrlExists($repo)) {
+                $local_repo = $this->repo_manager->getLocalRepoByUrl($repo->getUrl());
+            }
 
-                return Command::FAILURE;
+            $repo = $repo->withIsGlobal(true);
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Commands/Pack/ImportCommand.php]: done!

File: app/src/Commands/Repo/RepoManager.php (hunk 2 above)
Comment (daniwe4) [reply app/src/Commands/Repo/RepoManager.php:176]: done!

**Pull request merged**